mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentMapper.java

```diff
@@ -112,17 +112,16 @@ public InputStream loadAttachmentContent(AttachmentId attachmentId) throws Attac
 
     @Override
     public Mono<Attachment> storeAttachmentForOwner(String contentType, InputStream inputStream, Username owner) {
-        SizeInputStream sizeInputStream = new SizeInputStream(inputStream);
+        CurrentPositionInputStream currentPositionInputStream = new CurrentPositionInputStream(inputStream);
         AttachmentId attachmentId = AttachmentId.random();
-
         return ownerDAO.addOwner(attachmentId, owner)
-            .flatMap(any -> Mono.from(blobStore.save(blobStore.getDefaultBucketName(), sizeInputStream, LOW_COST)))
-            .map(blobId -> new DAOAttachment(attachmentId, blobId, contentType, sizeInputStream.getSize()))
+            .flatMap(any -> Mono.from(blobStore.save(blobStore.getDefaultBucketName(), currentPositionInputStream, LOW_COST)))
+            .map(blobId -> new DAOAttachment(attachmentId, blobId, contentType, currentPositionInputStream.getPosition()))
```

Comment:
this is adding a temporal coupling when using this CurrentPositionInputStream. How can we assure a good usage of it?

Reply:
Good question.

This is the best I (we - as part of the MOB) came up with.

A solution could be having the BlobStore returning you the size of the blob as a result of the save operation.

Please spare me that refactoring.

Reply:
maybe we can then having something more defensive: for example an `Optional<Integer> getSize()` which returns only when the stream is consumed. (with a good documentation stating why it can be empty)